Inferences/ForwardSubsumptionDemodulation.cpp

```diff
@@ -83,7 +83,16 @@ bool ForwardSubsumptionDemodulation::perform(Clause* cl, Clause*& replacement, C
 
   TIME_TRACE("forward subsumption demodulation");
 
+
   Ordering const& ordering = _salg->getOrdering();
+  // some assertions won't hold if we use an alasca ordering i.e. in an alasca ordering we might have s subterm t but s not less than t, especially for non-atomic theory terms. 
+  // e.g. a subterm a - a
+  // but  a greater 0 equiv a - a
+  // we use this boolean flag to disable the assertions broken in this case. 
+  // everything is still sound but might become incomplete.
+  // TODO investigate casese like this indepth and adjust demodulation appropriately
+  static bool isAlascaOrdering = env.options->termOrdering () == Shell::Options::TermOrdering::QKBO
```

Comment:
Given this remark, should there be a separate line or two in `Options::complete` pointing to Alasca ordering as a potential completeness compromiser? Or are we fine on non-arithmetic inputs (and thus "covered" through the `// general properties causing incompleteness` there)?

Reply:
For uninterpreted terms/literals the current alasca orderings will behave exactly like `kbo`. So that should be fine.